src/redux/slices/user.slice.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export interface UserState {
+  currentUser?: string; // username of logged in user
+}
+
+const initialState: UserState = {
+  currentUser: undefined,
+};
+
+const userSlice = createSlice({
+  name: "user",
+  initialState,
+  reducers: {
+    setCurrentUser(state, action) {
```

Comment:
Thoughts on typing the payload? action.type will also be auto generated by toolkit, so I've seen

`setCurrentUser(state, { payload }: PayloadAction<T>) {`

where T is the the payload type, which could be a primitive or an object, e.g. `{ id: string, name: string }`.
In this case T is just a string since you're assuming action.payload is the user. so

`setCurrentUser(state, { payload }: PayloadAction<string>) {`

Reply:
Yeah I'm happy to type it